src/builtins/Transformers.jl

```diff
@@ -28,9 +30,60 @@ const CONTINUOUS_ENCODER_DESCR = "Convert all `Finite` (categorical) and "*
 
 round_median(v::AbstractVector) = v -> round(eltype(v), median(v))
 
-_median       = e -> skipmissing(e) |> median
-_round_median = e -> skipmissing(e) |> (f -> round(eltype(f), median(f)))
-_mode         = e -> skipmissing(e) |> mode
+_median(e)       = skipmissing(e) |> median
+_round_median(e) = skipmissing(e) |> (f -> round(eltype(f), median(f)))
+_mode(e)         = skipmissing(e) |> mode
+
+@with_kw_noshow mutable struct UnivariateFillImputer <: Unsupervised
+    continuous_fill::Function = _median
+    count_fill::Function      = _round_median
+    finite_fill::Function     = _mode
+end
+
+function MLJBase.fit(transformer::UnivariateFillImputer,
+                      verbosity::Integer,
+                      v)
+
+    filler(v, ::Type) = throw(ArgumentError(
+        "Imputation is not supported for vectors "*
+        "of elscitype $(elscitype(v))."))
+    filler(v, ::Type{<:Union{Continuous,Missing}}) =
+        transformer.continuous_fill(v)
+    filler(v, ::Type{<:Union{Count,Missing}}) =
+        transformer.count_fill(v)
+    filler(v, ::Type{<:Union{Finite,Missing}}) =
+        transformer.finite_fill(v)
+
+    fitresult = (filler=filler(v, elscitype(v)),)
+    cache = nothing
+    report = nothing
+
+    return fitresult, cache, report
+
+end
+
+function MLJBase.transform(transformer::UnivariateFillImputer,
+                           fitresult,
+                           vnew)
+
+    filler = fitresult.filler
+    scitype(filler) <: elscitype(vnew) || error(
+    "Attempting to impute a value of scitype $(scitype(filler)) "*
+    "into a vector of incompatible elscitype, namely $(elscitype(vnew)). ")
+
+    if Missing <: elscitype(vnew)
+        w = copy(vnew) # transform must be non-mutating
+        w[ismissing.(w)] .= filler
+        w_tight = convert.(nonmissing(eltype(w)), w)
```

Comment:
@ablaom I know this has already been merged. I hope you don't mind me making a slight change here.
```julia
 w = copy(vnew) # transform must be non-mutating
        w[ismissing.(w)] .= filler
        w_tight = convert.(nonmissing(eltype(w)), w)
```
the above code has too much allocations (`w = copy(vnew) `  and  `w_tight = convert.(nonmissing(eltype(w)), w)`. Since a new array must always be created due the requirement that `transform` must be non-mutating , i feel that the following rewrite is slightly more efficient (Any slight increase in efficiency is required. right?)
```julia
w_tight = similar(vnew, nonmissing(eltype(vnew)))
 @inbounds for i in eachindex(vnew)
       ismissing(vnew[i]) ? (w_tight[i] = filler ) : (w_tight[i] = vnew[i])
     end
```
Maybe the following contrived example can help show why?
```julia
using Random, BenchmarkTools
function h1!(vnew, filler) #code avoiding double allocation
       w = similar(vnew, nonmissingtype(eltype(vnew)))
       @inbounds for i in eachindex(vnew)
       ismissing(vnew[i]) ? (w[i] = filler ) : (w[i] = vnew[i])
       end
       w
end
function h2!(vnew, filler) #code with double allocations
       w = copy(vnew) # transform must be non-mutating
               w[ismissing.(w)] .= filler
               w_tight = convert.(nonmissingtype(eltype(w)), w)
       w_tight
end
n = [repeat([missing],10000)..., rand(20000)...]; #array containing missing values
n1 = copy(n)
n2 = copy(n)
shuffle!(n);
n3 = copy(n);
n4 = copy(n);
julia> @btime h1!($n1, 0);
  38.010 μs (2 allocations: 234.45 KiB)
julia> @btime h2!($n2, 0);
  144.883 μs (9 allocations: 584.45 KiB)
julia> @btime h1!($n3, 0);
  166.380 μs (2 allocations: 234.45 KiB)

julia> @btime h2!($n4, 0);
  173.802 μs (9 allocations: 584.45 KiB)
```
`h1!` compared to `h2!` is slightly faster and involves little allocations (this becomes important if this code is called by other functions).

Reply:
Yes great idea! Can you make a new PR?

Reply:
And thank you for investigating.